### PR TITLE
Read all dependencies and return named list

### PR DESCRIPTION
**Pull request by jread-usgs**
for #91 and #104

@wdwatkins and @lindsaycarr review?

### CODE REVIEW

**Comment by wdwatkins** [comment]
I'll take a look later today

**Comment by lindsaycarr** [comment]
@jread-usgs how is this addressing #91?

File: tests/testthat/tests-readData.R [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+viz <- as.viz(list(
+  depends = c("MayflyNymph", "carData")
+))
+
+context("readData")
+
+test_that("readData works as expected", {
+  
+  mayfly <<- readData(viz[['depends']][1])
+  carData <<- readData(viz[['depends']][2])
```
Comment (lindsaycarr) [inline tests/testthat/tests-readData.R]: I always forget about this little trick - nice!

**Comment by lindsaycarr** [comment]
Looks good to me for addressing #104, but #91 is about R package dependencies not dependencies within the viz fetch-->process-->visualize pipeline.

**Comment by jread-usgs** [comment]
@lindsaycarr I meant https://github.com/USGS-VIZLAB/vizlab/issues/92 :+1:

File: tests/testthat/tests-readData.R [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+testtmp <- setup(copyTestViz=TRUE)
+viz <- as.viz(list(
+  depends = c("MayflyNymph", "carData")
+))
+
+context("readData")
+
+test_that("readData works as expected", {
+  
+  mayfly <<- readData(viz[['depends']][1])
```
Comment (wdwatkins) [inline tests/testthat/tests-readData.R]: I don't think so, either of those notations work when I load the `plot-cars` with `as.viz`.

File: tests/testthat/tests-readData.R (hunk 1 above)
Comment (jiwalker-usgs) [inline tests/testthat/tests-readData.R]: I think this bad form because tests are meant to be independent of each other.

File: R/readDepends.R [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  viz <- as.viz(viz)
+  readDepends(viz)
+}
+
+#' @rdname readDepends
+#' @export
+readDepends.viz <- function(viz){
+  viz.ids <- viz[['depends']]
+  depends <- lapply(viz.ids, readData)
+  names(depends) <- viz.ids
```
Comment (jiwalker-usgs) [inline R/readDepends.R]: I think this line should be `names(depends) <- names(viz.ids)`

**Pull request merged**